File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {

        int sLength = s.length();
        int tLength = t.length();
        
        if(sLength != tLength) {
            return false;
        }

        int[] alphabets = new int[26];

        for(int i = 0; i < sLength; i++) {
            alphabets[s.charAt(i) - 'a']++;
            alphabets[t.charAt(i) - 'a']--;
        }

        for(int i = 0; i < alphabets.length; i++) {
                if(alphabets[i] != 0) {
                    return false;
                }
        }
        
        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.kt
class Solution {
    fun isAnagram(s: String, t: String): Boolean {

        if(s.length != t.length){
            return false
        }

        val nums = IntArray(26)

        for(i in 0 until s.length) {
            nums[s[i] - 'a']++
            nums[t[i] - 'a']--
        }
        
        val isAnagram = nums.all {
            it == 0
        }

        return isAnagram

    }
}